ouroboros-consensus/test-util/Test/Util/Orphans/Arbitrary.hs

```diff
@@ -111,6 +129,129 @@ instance Arbitrary Seed where
 
     shrink = const []
 
+{-------------------------------------------------------------------------------
+  ConnectionPlan
+-------------------------------------------------------------------------------}
+
+-- | Inclusive bounds for each join delay in @'arbitrary' :: 'ConnectionPlan'@.
+boundsConnectionPlanDelay :: (Int, Int)
+boundsConnectionPlanDelay = (0, n - 1)
+{-
+
+  TODO how to explain/justify upper bound? Should it be higher?
+
+  n - 1 BFT passed 10000 tests
+
+  n BFT consistenly fails with at least the following
+
+      *** Failed! (after 560 tests and 1 shrink):
+      Exception:
+        withHandleModify: handle not found
+        CallStack (from HasCallStack):
+          error, called at src/Ouroboros/Storage/FS/Sim/MockFS.hs:294:9 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.FS.Sim.MockFS
+          withHandleModify, called at src/Ouroboros/Storage/FS/Sim/MockFS.hs:306:5 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.FS.Sim.MockFS
+          withHandleRead, called at src/Ouroboros/Storage/FS/Sim/MockFS.hs:341:5 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.FS.Sim.MockFS
+          withOpenHandleRead, called at src/Ouroboros/Storage/FS/Sim/MockFS.hs:509:26 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.FS.Sim.MockFS
+          hSeek, called at src/Ouroboros/Storage/FS/Sim/STM.hs:51:42 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.FS.Sim.STM
+          hSeek, called at src/Ouroboros/Storage/ImmutableDB/Impl.hs:1000:15 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.ImmutableDB.Impl
+          streamBinaryBlobsImpl, called at src/Ouroboros/Storage/ImmutableDB/Impl.hs:299:27 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.ImmutableDB.Impl
+          streamBinaryBlobs, called at src/Ouroboros/Storage/ImmutableDB/API.hs:199:5 in ouroboros-consensus-0.1.0.0-9EPQ2Vns1rz8WSl9f1afYD:Ouroboros.Storage.ImmutableDB.API
+      ShrunkConnectionPlan 1 (ConnectionPlan [4,5,4,0,1])
+
+  2 * n BFT consistently fails with at least the following
+
+      *** Failed! Exception: 'FailureException ExceptionInForkedThread (ImmDbParseFailure (Right (SlotNo {unSlotNo = 10})) (DeserialiseFailure 0 "expected list len"))' (after 6 tests and 7 shrinks):
```

Comment:
Also an unexpected error. Same for the other `ImmDbTrailingData`.

Reply:
Maybe the nodes are still busy at slot `n - 1` but are killed during a file-system operation?

Note to myself: we also want to test with corruption (and unexpected file-system errors?) in the future, so we might have to restart nodes during these tests.